lib/Common/Memory/Recycler.h

```diff
@@ -2353,6 +2385,7 @@ class ListTypeAllocatorFunc<Recycler, isLeaf>
 // Dummy class to choose the allocation function
 class RecyclerLeafAllocator;
 class RecyclerNonLeafAllocator;
+class RecyclerWriteBarrierAllocator;
```

Comment:
>RecyclerWriteBarrierAllocator [](start = 6, length = 29)

Could you explain why this new allocator type is needed and where it is used? This makes choosing a Recycler allocator more complex.